### PR TITLE
Smart Wild AI Option

### PR DESCRIPTION
**Pull request by ghoulslash**
- add flag to set smart wild AI
- default AI flags currently set to be based on the average level of the enemy side. See `GetWildAiFlags` in `src/battle_ai_main.c`
- a variable config option is introduced to enable adding more flags to the default ones

pokeemerald-0 pokeemerald-1

### CODE REVIEW

File: include/constants/battle_config.h [hunk 1]
```diff
@@ -198,6 +198,10 @@
 #define B_LAST_USED_BALL            TRUE        // If TRUE, the "last used ball" feature from Gen 7 will be implemented
 #define B_LAST_USED_BALL_BUTTON     R_BUTTON    // If last used ball is implemented, this button (or button combo) will trigger throwing the last used ball.
 
+// AI Settings
+#define B_SMART_WILD_AI_FLAG        0x1DE   // If not 0, you can set this flag in a script to enable smart wild pokemon
+#define B_VAR_WILD_AI_FLAGS         0x40F7  // If not 0, you can use this var to add to default wild AI flags. NOT usable with flags above (1 << 15)
```
Comment (AsparagusEduardo) [inline include/constants/battle_config.h:203]: These should be 0 by default imo.

Comment (AsparagusEduardo) [reply include/constants/battle_config.h:203]: Might also be good to move it near Flag and Var settings.

**Comment by ghoulslash** [comment]
Config values set to 0 by default and organized better. ready for re review

File: include/constants/battle_config.h [hunk 2]
```diff
@@ -176,11 +176,13 @@
 // Eg: Replace with FLAG_UNUSED_0x264 so you can use that flag to toggle the feature.
 #define B_FLAG_INVERSE_BATTLE       0     // If this flag is set, the battle's type effectiveness are inversed. For example, fire is super effective against water.
 #define B_FLAG_FORCE_DOUBLE_WILD    0     // If this flag is set, all land and surfing wild battles will be double battles.
+#define B_SMART_WILD_AI_FLAG        0   // If not 0, you can set this flag in a script to enable smart wild pokemon
```
Comment (LOuroboros) [inline include/constants/battle_config.h:179]:
```suggestion
#define B_SMART_WILD_AI_FLAG         0   // If not 0, you can set this flag in a script to enable smart wild pokemon
```

Comment (LOuroboros) [inline include/constants/battle_config.h:179]:
```suggestion
#define B_SMART_WILD_AI_FLAG        0    // If not 0, you can set this flag in a script to enable smart wild pokemon
```

**Review by AsparagusEduardo — APPROVED** [review]

**Pull request merged**